Nuro/Features/Child/Routine/Views/ChildRoutineView.swift

```diff
@@ -0,0 +1,170 @@
+        }
+    }
+    
+    private func setupCollectionView() {
+        activityCollectionView.dataSource = vc
+        
+        activityCollectionView.collectionViewLayout = collectionViewLayout()
+        activityCollectionView.isUserInteractionEnabled = false
+        
+        activityCollectionView.register(ChildActivityCollectionViewCell.self, forCellWithReuseIdentifier: ChildActivityCollectionViewCell.identifier)
+    }
+    
+    private func collectionViewLayout() -> UICollectionViewCompositionalLayout {
+        let item = NSCollectionLayoutItem(layoutSize: .init(widthDimension: .fractionalWidth(1), heightDimension: .fractionalHeight(1)))
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: .init(widthDimension: .absolute(Constants.HALF_SCREEN_WIDTH), heightDimension: .absolute(Constants.HALF_SCREEN_HEIGHT)), subitems: [item])
+        
+        let section = NSCollectionLayoutSection(group: group)
+        section.orthogonalScrollingBehavior = .groupPagingCentered
+        section.interGroupSpacing = 48
+        section.contentInsets = .init(top: 0, leading: 20, bottom: 0, trailing: 20)
+        
+        let layout = UICollectionViewCompositionalLayout(section: section)
+        return layout
+    }
+    
+    @objc func startActivity() {
+//        delegate.animateNextActivity()
+        
+        // TODO: Segue to full screen activity page
+        // ..
+    }
+    
+    func animateToNextActivity(totalActivity: Int, currTotalActivity: Int) {
+        UIView.animate(withDuration: 1, delay: 0, options: .curveEaseIn, animations: { [self] in
+            animateDeleteRow()
+            if totalActivity == currTotalActivity + 1 {
+                animateStickStartMovement()
+            }
+            else if currTotalActivity == 1 {
+                animateStickEndMovement()
+            }
+        }, completion: nil)
+    }
+    
+    func animateDeleteRow() {
+        activityCollectionView.performBatchUpdates {
+            activityCollectionView.deleteItems(at: [IndexPath(row: 0, section: 0)])
+        }
+    }
+    
+    func animateHideRow() {
+        activityCollectionView.reloadItems(at: [IndexPath(row: 0, section: 0)])
+    }
+    
+    func animateStickStartMovement() {
+        UIView.animate(withDuration: 1, delay: 0, options: .curveEaseIn, animations: { [self] in
+            stickView.frame = CGRect(x: stickView.frame.origin.x - Constants.HALF_SCREEN_WIDTH, y: stickView.frame.origin.y, width: stickView.frame.width, height: stickView.frame.height)
+        }, completion: { [self] _ in
+            stickView.snp.makeConstraints { make in
+                make.left.equalTo(self).offset(-24)
+            }
+        })
+    }
+    
+    func animateStickEndMovement() {
+        stickView.snp.removeConstraints()
+        stickView.snp.makeConstraints { make in
+            make.centerY.equalTo(self)
+            make.right.equalTo(self).offset(-80)
+            make.height.equalTo(64)
```

Comment:
Be carefull when giving a value to a view such as padding / height using constant value because with different screen sizes some constant value may not work well.  Ask the designer how the view behave as the screen changes and use a dynamic value